OpenSim/Simulation/Model/ExternalForce.cpp

```diff
@@ -160,27 +147,35 @@ void ExternalForce::extendConnectToModel(Model& model)
     _appliesTorque = appliesTorque();
 
     // hook up body pointers from names
-    if (_model){
-        _appliedToBody = 
-            static_cast<const PhysicalFrame*>(&_model->getComponent(appliedToBodyName));
-        _forceExpressedInBody = 
-            static_cast<const PhysicalFrame*>(&_model->getComponent(forceExpressedInBodyName));
-        _pointExpressedInBody = _specifiesPoint ? 
-            static_cast<const PhysicalFrame*>(&_model->getComponent(get_point_expressed_in_body()))
-            : nullptr;
+    _appliedToBody.reset();
+    _forceExpressedInBody.reset();
+    _pointExpressedInBody.reset();
+    if (_model) {
+        if (_model->hasComponent<PhysicalFrame>(appliedToBodyName))
+            _appliedToBody = &_model->getComponent<PhysicalFrame>(
+                    appliedToBodyName);
+        if (_model->hasComponent<PhysicalFrame>(forceExpressedInBodyName))
+            _forceExpressedInBody = &_model->getComponent<PhysicalFrame>(
+                    forceExpressedInBodyName);
+
+        if (_specifiesPoint && _model->hasComponent<PhysicalFrame>(
+                    get_point_expressed_in_body()))
+            _pointExpressedInBody = &_model->getComponent<PhysicalFrame>(
+                    get_point_expressed_in_body());
```

Comment:
This finding and connecting to dependent Components (e.g. `PhysicalFrame`s) is precisely what `Socket`s were intended for. If updating we might as well make it a proper Component.